compiler/rustc_hir_typeck/src/method/mod.rs

```diff
@@ -57,7 +57,7 @@ pub enum MethodError<'tcx> {
     PrivateMatch(DefKind, DefId, Vec<DefId>),
 
     // Found a `Self: Sized` bound where `Self` is a trait object.
-    IllegalSizedBound(Vec<DefId>, bool, Span),
+    IllegalSizedBound(Vec<DefId>, bool, Span, &'tcx hir::Expr<'tcx>),
```

Comment:
It's not immediately obvious that this is the self expression. Could you add a comment (or convert it into a variant with named fields) about that?